orangecontrib/spectroscopy/widgets/gui.py

```diff
@@ -474,3 +477,152 @@ def set_default(self, v):
+        self._boundingRect = None
+        self._endPoints = [0, 1]
+        self._bounds = None
+        self._lastViewSize = None
+
+        if label is None:
+            self.label = VerticalPeakLineLabel(self,
+                                               text=str(self.getXPos()), position=(1))
+
+    def setSpan(self, mn, mx):
+        if self.span != (mn, mx):
+            self.span = (mn, mx)
+            self.update()
+
+    def _computeBoundingRect(self):
+        vr = self.viewRect()
+        if vr is None:
+            return QtCore.QRectF()
+
+        px = self.pixelLength(direction=Point(1, 0), ortho=True)
+        if px is None:
+            px = 0
+        pw = max(self.pen.width() / 2, self.hoverPen.width() / 2)
+        w = max(4, 0 + pw) + 1
+        w = w * px
+        br = QtCore.QRectF(vr)
+        br.setBottom(-w)
+        br.setTop(w)
+        left = self.span[0]
+        right = self.span[1]
+        #sets our line span to go from our minimum value to maximum
+        br.setLeft(left)
+        br.setRight(right)
+        br = br.normalized()
+
+        vs = self.getViewBox().size()
+        if self.mouseDragEvent:
+            left = self.span[0] - self.getYPos()
+
+        if self._bounds != br or self._lastViewSize != vs:
+            self._bounds = br
+            self._lastViewSize = vs
+            self.prepareGeometryChange()
+
+        self._endPoints = (left, right)
+        self._lastViewRect = vr
+
+        return self._bounds
+
+    def boundingRect(self):
+        if self._boundingRect is None:
+            self._boundingRect = self._computeBoundingRect()
+        return self._boundingRect
+
+    def paint(self, p, *args):
+        p.setRenderHint(p.Antialiasing)
+        left, right = self._endPoints
+        pen = self.currentPen
+        pen.setJoinStyle(QtCore.Qt.MiterJoin)
+        p.setPen(pen)
+        p.drawLine(Point(left, 0), Point(right, 0))
+
+
+    def mouseDragEvent(self, ev):
+        if self.movable and ev.button() == QtCore.Qt.LeftButton:
+            if ev.isStart():
+                self.moving = True
+                self.cursorOffset = self.pos() - self.mapToParent(ev.buttonDownPos())
+                self.startPosition = self.pos()
+            ev.accept()
+
+            if not self.moving:
+                return
+
+            self.setPos(self.cursorOffset + self.mapToParent(ev.pos()))
+            self.sigDragged.emit(self)
+            self.updateLabel()
+            if ev.isFinish():
+                self.moving = False
+                self.sigPositionChangeFinished.emit(self)
+                self._computeBoundingRect()
+
+    def mouseClickEvent(self, ev):
+        if self.moving and ev.button() == QtCore.Qt.RightButton:
+            ev.accept()
+            self.setPos(self.startPosition)
+            self.moving = False
+            self.sigDragged.emit(self)
+            self.sigPositionChangeFinished.emit(self)
+
+    def mouseDoubleClickEvent(self, ev):
```

Comment:
Double clicking to remove is something we should not use. We need to stick with established UI paradigms.